File: src/java/magmac/app/compile/rule/locate/FirstLocator.java
package magmac.app.compile.rule.locate;

import magmac.api.None;
import magmac.api.Option;
import magmac.api.Some;

public class FirstLocator implements Locator {
    @Override
    public Option<Integer> locate(String input, String infix) {
        var index = input.indexOf(infix);
        if (-1 == index) {
            return new None<Integer>();
        }
        return new Some<>(index);
    }
}

File: src/web/magmac/app/compile/rule/locate/FirstLocator.ts
import { None } from "../../../../../magmac/api/None";
import { Option } from "../../../../../magmac/api/Option";
import { Some } from "../../../../../magmac/api/Some";
export class FirstLocator {
	locate(input : String, infix : String) : Option<Integer> {index : int=input.indexOf( infix);if(true){ return new None<Integer>( );;}return new Some<>( index);;}
}
